File: LeetCode/Easy/0771-jewels-and-stones/0771-jewels-and-stones.java
import java.util.HashMap;
import java.util.Map;

class Solution {
        public int numJewelsInStones(String jewels, String stones) {
            Map<Character, Integer> stonesMap = new HashMap<>();
            int result = 0;


            for(int i = 0; i < stones.length(); i++) {
                char stone = stones.charAt(i);
                stonesMap.put(stone, stonesMap.getOrDefault(stone, 0)+1);
            }

            for(int i = 0; i < jewels.length(); i++) {

                char jewel = jewels.charAt(i);
                if(stonesMap.get(jewel) != null) {
                    result += stonesMap.get(jewel);
                }
            }
            return result;
        }
    }

File: LeetCode/Easy/0771-jewels-and-stones/0771-jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        stones_map = {}
        result = 0

        for stone in stones:
            stones_map[stone] = stones_map.get(stone, 0) + 1
        
        for jewel in jewels:
            if jewel in stones_map:
                result += stones_map[jewel]

        return result